### ISSUE TITLE
doesn't daemonize

### ISSUE THREAD

**Issue by robmathews** [kigster/puma-daemon#9]
Not sure what I'm doing wrong here, but I've
* updated the Gemfile and bundle installed
* deployed

Then as a test, I go to the command line on the server and type: 
```
$ bundle exec pumad -C /mnt/code/shared/puma.rb config.ru
[19709] * Pruning Bundler environment
[19709] Puma starting in cluster mode...
[19709] * Puma version: 5.4.0 (ruby 3.0.2-p107) ("Super Flight")
[19709] *  Min threads: 20
[19709] *  Max threads: 20
[19709] *  Environment: staging
[19709] *   Master PID: 19709
[19709] *      Workers: 2
[19709] *     Restarts: (✔) hot (✔) phased
[19709] * Listening on unix:///mnt/code/shared/tmp/sockets/code-puma.sock
[19709] Use Ctrl-C to stop
```
The puma *is* working, but I'm not seeing the line 
```
Puma Daemon: Daemonizing (puma-daemon v0.1.1)...
```
and it also appears to be not daemonizing.

what am I missing?

# versions
puma-daemon 0.1.2
puma 5.0.4

**Comment by kigster** [comment]
Could you include your puma.rb and config.ru, assuming there is nothing confidential.

**Comment by robmathews** [comment]
config/puma.rb
```
require 'puma/daemon'

# The directory to operate out of.
# The default is the current directory.

directory '/mnt/code/current'

# Load “path” as a rackup file.
# The default is “config.ru”.

rackup '/mnt/code/current/config.ru'

# Set the environment in which the rack's app will run. The value must be a string.
# The default is “development”.

environment Rails.env

# Daemonize the server into the background. Highly suggest that
# this be combined with “pidfile” and “stdout_redirect”.
# The default is “false”.


# Store the pid of the server in the file at “path”.

pidfile '/mnt/code/current/tmp/pids/puma.pid'

# Use “path” as the file to store the server info state. This is
# used by “pumactl” to query and control the server.
state_path '/mnt/code/current/tmp/pids/puma.state'

# Redirect STDOUT and STDERR to files specified. The 3rd parameter
# (“append”) specifies whether the output is appended, the default is
# “false”.
stdout_redirect '/mnt/code/shared/log/puma_access.log', '/mnt/code/shared/log/puma_error.log', true

# Disable request logging.
# The default is “false”.

quiet

# Configure “min” to be the minimum number of threads to use to answer
# requests and “max” the maximum.
# The default is “0, 16”.

threads 0, 16

# Bind the server to “url”. “tcp://”, “unix://” and “ssl://” are the only
# accepted protocols.
# The default is “tcp://0.0.0.0:9292”.

bind 'unix:///mnt/code/shared/tmp/sockets/code-puma.sock'
# Listens on port 7001
# The default is 9292
#port 7001

# preload_app!

daemonize true
# Instead of “bind 'ssl://127.0.0.1:9292?key=path_to_key&cert=path_to_cert'” you
# can also use the “ssl_bind” option.

 # ssl_bind '127.0.0.1', '9292', { key: path_to_key, cert: path_to_cert }

# Code to run before doing a restart. This code should
# close log files, database connections, etc.

# This can be called multiple times to add code each time.

on_restart do
  puts 'On restart...'
end

# Command to use to restart puma. This should be just how to
# load puma itself (ie. 'ruby -Ilib bin/puma'), not the arguments
# to puma, as those are the same as the original process.

restart_command 'ruby -Ilib bin/puma'

# === Cluster mode ===

# How many worker processes to run.
# The default is “0”.

workers 2

# Code to run when a worker boots to setup the process before booting
# the app.
# This can be called multiple times to add hooks.

# === Puma control rack application ===

# Start the puma control rack application on “url”. This application can
# be communicated with to control the main server. Additionally, you can
# provide an authentication token, so all requests to the control server
# will need to include that token as a query parameter. This allows for
# simple authentication.

# Check out https://github.com/puma/puma/blob/master/lib/puma/app/status.rb
# to see what the app has available.

#activate_control_app 'unix:///var/run/pumactl.sock'
```
and config.ru
```
# This file is used by Rack-based servers to start the application.

require ::File.expand_path('../config/environment',  __FILE__)
run CODE::Application
```

**Comment by kigster** [comment]
Thanks for the context! Looks like maybe another bug. I've been really busy with work lately so I might  not get to look at for a couple of weeks.

**Comment by Sivard** [comment]
@robmathews Hello.
Try to comment string with `environment Rails.env`.
If I run local puma and add a string to the config, then I get a error `uninitialized constant #<Class:#<Puma::DSL:0x0000000001410bf8>>::Rails (NameError)`
You need to add another way of defining the environment in the config

**Comment by kigster** [comment]
This helps a lot!

**Comment by kigster** [comment]
Hopefully fixed with #11
